app/uk/gov/hmrc/customerprofile/connector/ApplePassConnector.scala

```diff
@@ -19,29 +19,53 @@ package uk.gov.hmrc.customerprofile.connector
 import com.google.inject.Inject
 import com.google.inject.name.Named
 import play.api.Logger
-import uk.gov.hmrc.customerprofile.domain.{RetrieveApplePass, GetApplePass, ApplePassUUIDGenerator}
+import play.api.http.Status.OK
+import play.api.libs.json.Format.GenericFormat
+import play.api.libs.json.{JsValue, Json}
+import uk.gov.hmrc.customerprofile.domain.{ApplePassIdGenerator, RetrieveApplePass}
 import uk.gov.hmrc.domain.Nino
-import uk.gov.hmrc.http.{CoreGet, CorePost, HeaderCarrier}
+import uk.gov.hmrc.http
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpException, HttpResponse}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class ApplePassConnector @Inject()(httpPost: CorePost, httpGet : CoreGet, @Named("find-my-nino-add-to-wallet") findMyNinoAddToWalletUrl: String){
+class ApplePassConnector @Inject()(http: HttpClient,  @Named("find-my-nino-add-to-wallet") findMyNinoAddToWalletUrl: String){
 
   val logger: Logger = Logger(this.getClass)
+  private val headers: Seq[(String, String)] = Seq("Content-Type" -> "application/json")
 
-  def createApplePass(nino : Nino, name: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[GetApplePass]  = {
-    httpPost.POST[ApplePassUUIDGenerator,GetApplePass](url = s"$findMyNinoAddToWalletUrl/find-my-nino-add-to-wallet/create-apple-pass", ApplePassUUIDGenerator(name, nino)) recover {
-      case e =>
-        logger.info(s"Error: ${e.getMessage}")
-        throw e
-    }
+  def createApplePass(nino: Nino, fullName: String)
+                     (implicit ec: ExecutionContext, headerCarrier: HeaderCarrier): Future[Some[String]] = {
```

Comment:
This should return a Future[Option[String]], but I don't see a need for it to be optional at all, so should really just be Future[String]